azure-iot-device/azure/iot/device/common/mqtt_transport.py

```diff
@@ -381,12 +383,16 @@ def connect(self, password=None):
             if self._websockets:
                 logger.info("Connect using port 443 (websockets)")
                 rc = self._mqtt_client.connect(
-                    host=self._hostname, port=443, keepalive=DEFAULT_KEEPALIVE
+                    host=self._hostname,
+                    port=443,
+                    keepalive=self._keep_alive if self._keep_alive else DEFAULT_KEEPALIVE,
                 )
             else:
                 logger.info("Connect using port 8883 (TCP)")
                 rc = self._mqtt_client.connect(
-                    host=self._hostname, port=8883, keepalive=DEFAULT_KEEPALIVE
+                    host=self._hostname,
+                    port=8883,
+                    keepalive=self._keep_alive if self._keep_alive else DEFAULT_KEEPALIVE,
```

Comment:
>keepalive=self._keep_alive if self._keep_alive else DEFAULT_KEEPALIVE, [](start = 20, length = 70)

Another advantage to moving the Default keepalive to the config is that we get rid of this conditional logic here. MQTTTransport will take whatever value is in the config, and doesn't need branching logic.